weasyl/controllers/general.py

```diff
@@ -1,142 +1,136 @@
+        }
+
+        search_query = search.Query.parse(meta["q"], find)
 
+        if search_query.find == "user":
+            query = search.select_users(meta["q"])
+            next_count = back_count = 0
+        else:
+            search_query.ratings.update(ratings.CHARACTER_MAP[rating_code].code for rating_code in meta["rated"])
+
+            query, next_count, back_count = search.select(
+                userid=request.userid,
+                rating=rating,
+                limit=63,
+                search=search_query,
+                within=meta["within"],
+                cat=meta["cat"],
+                subcat=meta["subcat"],
+                backid=meta["backid"],
+                nextid=meta["nextid"])
+
+        page.append(define.render("etc/search.html", [
+            # Search method
+            {"method": "search"},
+            # Search metadata
+            meta,
+            # Search results
+            query,
+            next_count,
+            back_count,
+            # Submission subcategories
+            macro.MACRO_SUBCAT_LIST,
+            search.COUNT_LIMIT,
+        ]))
+    elif form.find:
+        query = search.browse(request.userid, rating, 66, form)
+
+        meta = {
+            "find": form.find,
+            "cat": int(form.cat) if form.cat else None,
+        }
 
-class popular_(controller_base):
-    def GET(self):
-        return define.webpage(self.user_id, 'etc/popular.html', [
-            list(itertools.islice(
-                index.filter_submissions(self.user_id, submission.select_recently_popular(), incidence_limit=1), 66))
-        ])
+        page.append(define.render("etc/search.html", [
+            # Search method
+            {"method": "browse"},
+            # Search metadata
+            meta,
+            # Search results
+            query,
+            0,
+            0,
+        ]))
+    else:
+        page.append(define.render("etc/search.html", [
+            # Search method
+            {"method": "summary"},
+            # Search metadata
+            None,
+            # Search results
+            {
+                "submit": search.browse(self.user_id, rating, 22, form, find="submit"),
+                "char": search.browse(self.user_id, rating, 22, form, find="char"),
+                "journal": search.browse(self.user_id, rating, 22, form, find="journal"),
+            },
+        ]))
+
+    return Response(define.common_page_end(request.userid, page, rating, options={'search'}))
+
+
+def streaming_(request):
+    extras = {
+        "title": "Streaming",
+    }
+    rating = define.get_rating(request.userid)
+    return Response(define.webpage(request.userid, 'etc/streaming.html',
+                                   [profile.select_streaming(request.userid, rating, 300, order_by="start_time desc")],
+                                   **extras))
+
+
+def site_update_(request):
+    # TODO: Handle a non-integer with something better than a generic error?
```

Comment:
We can use a regular expression in the pattern and let it default to a 404, right?